client/jetpack-app/plans/main.jsx

```diff
@@ -0,0 +1,109 @@
+import { getPlan, PLAN_FREE } from '@automattic/calypso-products';
+import { translate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import QueryPlans from 'calypso/components/data/query-plans';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { LoadingEllipsis } from 'calypso/components/loading-ellipsis';
+import Main from 'calypso/components/main';
+import { getPlanCartItem } from 'calypso/lib/cart-values/cart-items';
+import PlansFeaturesMain from 'calypso/my-sites/plans-features-main';
+import { getPlanSlug } from 'calypso/state/plans/selectors';
+
+import './style.scss';
+
+const Header = ( { paidDomainName } ) => (
+	<div className="plans__header">
+		<FormattedHeader
+			brandFont
+			headerText={ translate( 'Choose the perfect plan' ) }
+			align="center"
+		/>
+		{ paidDomainName ? (
+			<>
+				<p>
+					{ translate(
+						'With your annual plan, you’ll get %(domainName)s {{strong}}free for the first year{{/strong}}.',
+						{
+							args: {
+								domainName: paidDomainName,
+							},
+							components: { strong: <strong /> },
+						}
+					) }
+				</p>
+				<p>
+					{ translate(
+						'You’ll also unlock advanced features that make it easy to build and grow your site.'
+					) }
+				</p>
+			</>
+		) : (
+			<p>{ translate( 'See and compare the features available on each WordPress.com plan.' ) }</p>
+		) }
+	</div>
+);
+
+const Plans = ( { paidDomainName, onUpgradeClick, handleRedirect } ) => (
+	<PlansFeaturesMain
+		paidDomainName={ paidDomainName }
+		intent="plans-jetpack-app-site-creation"
+		isInSignup={ true }
+		intervalType="yearly"
+		onUpgradeClick={ onUpgradeClick }
+		plansWithScroll={ false }
+		removePaidDomain={ handleRedirect }
+		hidePlanTypeSelector={ true }
+		hidePlansFeatureComparison={ true }
+	/>
+);
+
+const Loading = () => (
+	<div className="plans__loading">
+		<LoadingEllipsis active />
+	</div>
+);
```

Comment:
Unnecessary sub component. Don't be afraid of a large JSX tree. There is no reason to move parts of the tree into smaller components unless there is some data related complexity or until you sense the need to reuse a component elsewhere. Just keep the tree inline as much as possible. This way it's much easier to reason about the code without haven't to click through multiple times to reach the changed piece of code.